### PR TITLE
Make geocoding return multiple matches instead of just the best

### PR DESCRIPTION
**Pull request by Ef55**

### CODE REVIEW

File: app/src/main/java/com/github/onedirection/geocoding/GeocodingService.java [hunk 1]
```diff
@@ -16,7 +16,15 @@
      * @return The coordinates if the location is found.
      */
     default CompletableFuture<Coordinates> getBestCoordinates(String locationName){
```
Comment (Flechman) [inline app/src/main/java/com/github/onedirection/geocoding/GeocodingService.java:18]: Shouldn't you need to check for null ?

File: app/src/main/java/com/github/onedirection/geocoding/GeocodingService.java [hunk 2]
```diff
@@ -16,7 +16,15 @@
...
-        return getBestNamedCoordinates(locationName).thenApply(p -> p.dropName());
+        return getBestNamedCoordinates(locationName).thenApply(NamedCoordinates::dropName);
+    }
+
+    /**
+     * @param locationName Some location's name.
+     * @return A list of its possible coordinates.
+     */
+    default CompletableFuture<List<Coordinates>> getCoordinates(String locationName, int count){
```
Comment (Flechman) [inline app/src/main/java/com/github/onedirection/geocoding/GeocodingService.java:26]: Check for null ?

File: app/src/main/java/com/github/onedirection/utils/Monads.java [hunk 3]
```diff
@@ -16,13 +19,17 @@ private Monads() {}
 
     public static <T> CompletableFuture<T> toFuture(Task<T> m){
         CompletableFuture<T> result = new CompletableFuture<>();
-        m.addOnSuccessListener(r -> result.complete(r))
-            .addOnFailureListener(t -> result.completeExceptionally(t))
+        m.addOnSuccessListener(result::complete)
```
Comment (Flechman) [inline app/src/main/java/com/github/onedirection/utils/Monads.java:22]: Scala-like ❤️

**Review by Flechman — APPROVED** [review]
Very clean and concise code

File: app/src/main/java/com/github/onedirection/geocoding/GeocodingService.java (hunk 2 above)
Comment (Ef55) [reply app/src/main/java/com/github/onedirection/geocoding/GeocodingService.java:26]: There is no need here; the argument is just passed to another function.

File: app/src/main/java/com/github/onedirection/geocoding/GeocodingService.java (hunk 1 above)
Comment (Ef55) [reply app/src/main/java/com/github/onedirection/geocoding/GeocodingService.java:18]: There is no need here; the argument is just passed to another function.

**Review by TheImpressiveDonut — APPROVED** [review]
Looks good to me

**Pull request merged**